### PR TITLE
Feature/user inheritance

### PR DESCRIPTION
**Pull request by Carlangasv**

### CODE REVIEW

File: src/main/java/perficient/academic/universityapplication/models/Student.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import perficient.academic.universityapplication.enums.UserType;
+
+@Entity
+@NoArgsConstructor
+@PrimaryKeyJoinColumn(name = "id")
+public class Student extends User
+{
+	@NotNull
+	@Getter
+	@Setter
```
Comment (sf-burgos) [inline src/main/java/perficient/academic/universityapplication/models/Student.java]: Why are you using those annotations here ?

File: src/main/java/perficient/academic/universityapplication/models/User.java [hunk 2]
```diff
@@ -1,34 +1,49 @@
...
-@Getter
-@Setter
-@NoArgsConstructor
+@Data
 @AllArgsConstructor
-@Table(name = "users")
+@NoArgsConstructor
+@Inheritance(strategy = InheritanceType.JOINED)
+@Table(name = "my_user", uniqueConstraints = @UniqueConstraint(columnNames = { "governmentId", "email" }))
 public class User
```
Comment (sf-burgos) [inline src/main/java/perficient/academic/universityapplication/models/User.java:24]: Do you need getters and setters for this class?

File: src/main/java/perficient/academic/universityapplication/repositories/StudentRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import perficient.academic.universityapplication.models.Student;
+
+public interface StudentRepository extends JpaRepository<Student, Long>
+{
+	Student getStudentByEmail(String email);
+
+	Student getStudentByGovernmentId(Long governmentId);
+
+	void removeByGovernmentId(Long governmentId);
```
Comment (sf-burgos) [inline src/main/java/perficient/academic/universityapplication/repositories/StudentRepository.java:12]: Excelent remove here, good work man!

File: src/main/java/perficient/academic/universityapplication/repositories/TeacherRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package perficient.academic.universityapplication.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import perficient.academic.universityapplication.models.Teacher;
+
+
+public interface TeacherRepository extends JpaRepository<Teacher, Long>
```
Comment (sf-burgos) [inline src/main/java/perficient/academic/universityapplication/repositories/TeacherRepository.java:7]: Probably for the next week create some functionalities here!

**Review by sf-burgos — APPROVED** [review]

File: src/main/java/perficient/academic/universityapplication/models/Student.java (hunk 1 above)
Comment (Carlangasv) [reply src/main/java/perficient/academic/universityapplication/models/Student.java]: It was a mistake, those @Getter and @Setter should not be there, I'm going to remove them

**Pull request merged**